source/common/network/connection_impl.cc

```diff
@@ -706,6 +706,40 @@ void ConnectionImpl::flushWriteBuffer() {
   }
 }
 
+ServerConnectionImpl::ServerConnectionImpl(Event::Dispatcher& dispatcher,
+                                           ConnectionSocketPtr&& socket,
+                                           TransportSocketPtr&& transport_socket,
+                                           StreamInfo::StreamInfo& stream_info, bool connected)
+    : ConnectionImpl(dispatcher, std::move(socket), std::move(transport_socket), stream_info,
+                     connected) {}
+
+void ServerConnectionImpl::setTransportSocketConnectTimeout(std::chrono::milliseconds timeout) {
+  if (!transport_connect_pending_) {
+    return;
+  }
+  if (transport_socket_connect_timer_ == nullptr) {
+    transport_socket_connect_timer_ =
+        dispatcher_.createTimer([this] { onTransportSocketConnectTimeout(); });
+  }
+  transport_socket_connect_timer_->enableTimer(timeout);
+}
+
+void ServerConnectionImpl::raiseEvent(ConnectionEvent event) {
+  switch (event) {
+  case ConnectionEvent::Connected:
+  case ConnectionEvent::RemoteClose:
+  case ConnectionEvent::LocalClose:
```

Comment:
Do you envision other events in the future in which we wouldn't do this? Seems like you can just remove the switch and do this always?

Reply:
I don't know, but this seemed like a good way of 1) demonstrating that this is supposed to handle all cases identically (i.e. I didn't consider one and forget about the others) and 2) ensuring that if someone ever adds an event type, they have to consider how to modify this since it'll cause a compile error otherwise.

Reply:
OK fair enough. Fine by me.